kiwi_lint/similar_string.py

```diff
@@ -0,0 +1,74 @@
+import string
+
+import astroid
+from pylint.checkers import BaseChecker, utils
+from pylint.interfaces import IAstroidChecker
+from textdistance import levenshtein
+
+
+class SimilarStringChecker(BaseChecker):
+    __implements__ = (IAstroidChecker,)
+    name = "similar-string-checker"
+
+    msgs = {
+        "R5011": (
+            '"%s" is similar to "%s". Try keeping them the same to reduce work for translators!',
+            "similar-string",
+            "Similar strings should be avoided for single translation",
+        )
+    }
+
+    _dict_of_strings = {}
+    threshold = 0.8
+
+    @staticmethod
+    def clean_string(text):
+        """
+        This method removes the operators and other punctuations
+        used in the string to avoid unwanted data to add up to
+        the similarity between the strings.
+        """
+        for word in text:
+            if word not in string.punctuation:
+                text += "".join(word)
```

Comment:
You don't need to use `.join()`, concatenation to `text` produces the same result.

Reply:
However modifying the input parameter while looping over it is not a good idea. Moreover the results are incorrect.

```
<<< input: NEW: Bug #%(pk)d - %(summary)s
>>> output: NEW: Bug #%(pk)d - %(summary)sNEW Bug pkd  summarys
```

Reply:
should I leave '(' and ')' while checking for punctuation?

Reply:
don't change the if condition for now, change how the method collects its result.